src/SearchHeuristics/index.ts

```diff
@@ -0,0 +1,38 @@
+import { Document } from '../SearchIndex';
+
+// A series of basic heuristics that we can apply to determine if a search result should be indexed
+// TODO: Document interface should be defined here and this should be a document abstraction?
+// or a SearchableDocument abstraction?
+
+interface FilteredDocuments {
+  searchable: Document[];
+  unsearchable: Document[];
+}
+
+const hasMinimumTextThreshold = (doc: Document) => {
+  const MINIMUM_CONTENT_LENGTH = 40;
+  return doc.text.length >= MINIMUM_CONTENT_LENGTH;
+};
+
+const hasPreview = (doc: Document) => !!doc.preview?.trim();
+
+const isInteractive = (doc: Document) => doc.links?.length >= 2;
+
+const meetsMinimumSearchableStandard = (doc: Document) => {
+  const failsThreshold = !hasPreview(doc) && (!hasMinimumTextThreshold(doc) || !isInteractive(doc));
```

Comment:
Preliminary querying of our production index shows this will remove roughly 367 results - without the `isInteractive` check we'd remove 2k results. 
```{preview: '','links.2': {$exists:true},$expr: {$gt: [ { $strLenCP: '$text' }, 40]}}``` 

For now, I'm erring on the side of tunable caution.